docs/noise.ipynb

```diff
@@ -62,196 +62,673 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "8143ae0c7a34"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Example of Monte Carlo wavefunction simulation with the `run` method.\"\"\"\n",
+    "circuit = cirq.Circuit(\n",
+    "    cirq.bit_flip(p=0.5).on(qbit),\n",
+    "    cirq.measure(qbit),\n",
+    ")\n",
+    "res = sim.run(circuit, repetitions=100)\n",
+    "print(res.histogram(key=qbit))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "0f123f0e3c55"
+   },
+   "source": [
+    "### Adding noise to circuits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "473ac025a226"
+   },
+   "source": [
+    "Often circuits are defined with just unitary operations, but we want to simulate them with noise. There are several methods for inserting noise in Cirq."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "d0de2c475f12"
+   },
+   "source": [
+    "For any circuit, the `with_noise` method can be called to insert a channel after every moment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "568530c8707b"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"One method to insert noise in a circuit.\"\"\"\n",
+    "# Define some noiseless circuit.\n",
+    "circuit = cirq.testing.random_circuit(\n",
+    "    qubits=3, n_moments=3, op_density=1, random_state=11\n",
+    ")\n",
+    "\n",
+    "# Display the noiseless circuit.\n",
+    "print(\"Circuit without noise:\")\n",
+    "print(circuit)\n",
+    "\n",
+    "# Add noise to the circuit.\n",
+    "noisy = circuit.with_noise(cirq.depolarize(p=0.01))\n",
+    "\n",
+    "# Display it.\n",
+    "print(\"\\nCircuit with noise:\")\n",
+    "print(noisy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "c2206a7b7dd4"
+   },
+   "source": [
+    "This circuit can then be simulated using the methods described above."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "11b18c640767"
+   },
+   "source": [
+    "Another technique is to pass a noise channel to the density matrix simulator as shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "3d16924842bf"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Define a density matrix simulator with a noise model.\"\"\"\n",
+    "noisy_dsim = cirq.DensityMatrixSimulator(\n",
+    "    noise=cirq.generalized_amplitude_damp(p=0.1, gamma=0.5)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "99ac77a2c896"
+   },
+   "source": [
+    "Other than these general methods, channels can be added to circuits at any moment just as gates are. The channels can be different, be correlated, act on a subset of qubits, be custom defined, etc."
```

Comment:
If I remember rightly, it should be possible to generate a 'noisy_moment' function that takes in a moment and adds noise to the entire chip. This could be in principle useful (e.g. if one wants to simulate crosstalk by adding unwanted rotations to nearby gates). Can we include mention + perhaps an example of this?

Reply:
Do you mean the `Circuit.with_noise` method (example in previous cell) or something else?

Reply:
I mean the NoiseModel.noisy_moment and NoiseModel.noisy_moments methods. These can be quite useful in some cases (e.g. the example I gave).